src/metabase/driver/hive_like.clj

```diff
@@ -0,0 +1,173 @@
+  {;; Numeric types
+   :tinyint :type/Integer
+   :smallint :type/Integer
+   :int :type/Integer
+   :integer :type/Integer
+   :bigint :type/BigInteger
+   :float :type/Float
+   :double :type/Float
+   (keyword "double precision") :type/Float
+   :decimal :type/Decimal
+   ;; Date/Time types
+   :timestamp :type/DateTime
+   :date :type/Date
+   :interval :type/*
+   :string :type/Text
+   :varchar :type/Text
+   :char :type/Text
+   :boolean :type/Boolean
+   :binary :type/*})
+
+(def ^:const now (hsql/raw "NOW()"))
+
+(defn unix-timestamp->timestamp [expr seconds-or-milliseconds]
+  (hx/->timestamp
+   (hsql/call :from_unixtime (case seconds-or-milliseconds
+                               :seconds      expr
+                               :milliseconds (hx// expr 1000)))))
+
+(defn- date-format [format-str expr]
+  (hsql/call :date_format expr (hx/literal format-str)))
+
+(defn- str-to-date [format-str expr]
+  (hx/->timestamp
+   (hsql/call :from_unixtime
+              (hsql/call :unix_timestamp
+                         expr (hx/literal format-str)))))
+
+(defn- trunc-with-format [format-str expr]
+  (str-to-date format-str (date-format format-str expr)))
+
+(defn date [unit expr]
+  (case unit
+    :default expr
+    :minute (trunc-with-format "yyyy-MM-dd HH:mm" (hx/->timestamp expr))
+    :minute-of-hour (hsql/call :minute (hx/->timestamp expr))
+    :hour (trunc-with-format "yyyy-MM-dd HH" (hx/->timestamp expr))
+    :hour-of-day (hsql/call :hour (hx/->timestamp expr))
+    :day (trunc-with-format "yyyy-MM-dd" (hx/->timestamp expr))
+    :day-of-week (hx/->integer (date-format "u"
+                                            (hx/+ (hx/->timestamp expr)
+                                                  (hsql/raw "interval '1' day"))))
+    :day-of-month (hsql/call :dayofmonth (hx/->timestamp expr))
+    :day-of-year (hx/->integer (date-format "D" (hx/->timestamp expr)))
+    :week (hsql/call :date_sub
+                     (hx/+ (hx/->timestamp expr)
+                           (hsql/raw "interval '1' day"))
+                     (date-format "u"
+                                  (hx/+ (hx/->timestamp expr)
+                                        (hsql/raw "interval '1' day"))))
+    :week-of-year (hsql/call :weekofyear (hx/->timestamp expr))
+    :month (hsql/call :trunc (hx/->timestamp expr) (hx/literal :MM))
+    :month-of-year (hsql/call :month (hx/->timestamp expr))
+    :quarter (hsql/call :add_months
+                        (hsql/call :trunc (hx/->timestamp expr) (hx/literal :year))
+                        (hx/* (hx/- (hsql/call :quarter (hx/->timestamp expr))
+                                    1)
+                              3))
+    :quarter-of-year (hsql/call :quarter (hx/->timestamp expr))
+    :year (hsql/call :year (hx/->timestamp expr))))
+
+(defn date-interval [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s)" (int amount) (name unit))))
+
+(defn string-length-fn [field-key]
```

Comment:
As with other namespaces, functions in this one should be private if possible otherwise they need to be documented

Reply:
I don't know if you saw it [in the Developers' Guide](https://github.com/metabase/metabase/blob/master/docs/developers-guide.md#run-the-linters) but you can and should run the liners locally:

```bash
lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter
```

That will catch these sorts of things and save us both some back-and-forth on the PR

Reply:
Sorry, I may have missed that one, or it was added after the last time I looked at the Developer's Guide. My latest patch set passes the linters.